client/app/controllers/subscribe.js

```diff
@@ -0,0 +1,92 @@
+import Controller from '@ember/controller';
+import { action } from '@ember/object';
+import fetch from 'fetch';
+import ENV from 'labs-zap-search/config/environment';
+
+const tlds = ['.com', '.gov', '.edu', '.net', '.org'];
+
+export default class SubscribeController extends Controller {
+  lastEmailChecked = '';
+
+  emailAlreadyExists = false;
+
+  emailNeedsConfirmation = false;
+
+  startContinuouslyChecking = false;
+
+  emailSent = false;
+
+
+  @action
+  async checkExistingEmail(event) {
+    const email = event.target.value;
+    if (email === this.lastEmailChecked) { return; }
+    this.set('lastEmailChecked', email);
+    this.set('startContinuouslyChecking', true);
+
+    try {
+      const response = await fetch(`${ENV.host}/subscribers/email/${email}`);
+      const userData = await response.json();
+
+      if (userData.error) {
+        this.set('emailAlreadyExists', false);
+        this.set('emailNeedsConfirmation', false);
+        this.set('emailSent', false);
+        return;
+      }
+
+      if (userData.message === 'User already subscribed.') {
```

Comment:
I had Pratishta update the response from this API to include a boolean `confirmed` value. Let's use that instead of comparing against magic strings.